Directory.Build.props

```diff
@@ -9,6 +9,8 @@
     <Nullable>enable</Nullable>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <GenerateDocumentationFile>true</GenerateDocumentationFile>
+    <EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>
+    <TreatWarningsAsErrors Condition="'$(Configuration)' == 'Release'">true</TreatWarningsAsErrors>
```

Comment:
> We treat warnings as errors in the official build already. 

I looked for this and didn't find it, but cool. I'll remove this then because it's redundant.

> In local dev builds we want them to remain just warnings so they don't slow down devs "inner loop" of code-build-test.

Yes, that is achieved by this line.

Reply:
FYI the official builds do it via this line:

https://github.com/microsoft/CsWin32/blob/428710543b78f20eb15c9a007cbce2224e556e3a/azure-pipelines.yml#L21